CMakeLists.txt

```diff
@@ -130,28 +130,83 @@ if (WIN32)
     endif()
 else()
     if (APPLE)
-        target_link_libraries (OpenCL ${CMAKE_THREAD_LIBS_INIT})
+        target_link_libraries (OpenCL PRIVATE ${CMAKE_THREAD_LIBS_INIT})
     else ()
         set_target_properties (OpenCL PROPERTIES LINK_FLAGS "-Wl,--version-script -Wl,${CMAKE_CURRENT_BINARY_DIR}/loader/linux/icd_exports.map")
-        target_link_libraries (OpenCL ${CMAKE_THREAD_LIBS_INIT})
+        target_link_libraries (OpenCL PRIVATE ${CMAKE_THREAD_LIBS_INIT})
     endif ()
 endif ()
 
-include_directories (${OPENCL_ICD_LOADER_HEADERS_DIR})
+find_package(OpenCLHeaders REQUIRED)
+if (TARGET OpenCL::OpenCLHeaders)
+  target_link_libraries (OpenCL PUBLIC OpenCL::OpenCLHeaders)
+  link_libraries(OpenCL::OpenCLHeaders)
+else ()
+  include_directories (${OPENCL_ICD_LOADER_HEADERS_DIR})
+endif ()
 if (ENABLE_OPENCL30_PROVISIONAL)
   add_definitions (-DCL_TARGET_OPENCL_VERSION=300)
 else()
   add_definitions (-DCL_TARGET_OPENCL_VERSION=220)
 endif()
 
-target_include_directories (OpenCL PRIVATE ${CMAKE_CURRENT_BINARY_DIR} loader)
-target_link_libraries (OpenCL ${CMAKE_DL_LIBS})
+add_library (OpenCL::OpenCLICDLoader
+  ALIAS OpenCL
+)
+
+target_include_directories (OpenCL
+  PRIVATE
+    ${CMAKE_CURRENT_BINARY_DIR}
+    loader
+)
+target_link_libraries (OpenCL PUBLIC ${CMAKE_DL_LIBS})
 
 include (CTest)
 if (BUILD_TESTING)
     add_subdirectory (test)
 endif()
 
+install(
+  TARGETS OpenCL
+  EXPORT OpenCLICDLoaderTargets
+)
+
+include(GNUInstallDirs)
```

Comment:
We do this on line 4, is the repeated `include` necessary?

Reply:
No, removed the duplicate.